tools/cldr-code/src/main/java/org/unicode/cldr/util/NameType.java

```diff
@@ -0,0 +1,275 @@
+package org.unicode.cldr.util;
+
+import java.util.Locale;
+
+public enum NameType {
+    NONE,
+    LANGUAGE,
+    SCRIPT,
+    TERRITORY,
+    VARIANT,
+    CURRENCY,
+    CURRENCY_SYMBOL,
+    TZ_EXEMPLAR,
+    TZ_GENERIC_LONG,
+    TZ_GENERIC_SHORT,
+    TZ_STANDARD_LONG,
+    TZ_STANDARD_SHORT,
+    TZ_DAYLIGHT_LONG,
+    TZ_DAYLIGHT_SHORT,
+    KEY,
+    KEY_TYPE,
+    SUBDIVISION;
+
+    // The order of rows must correspond to INDEX_LANGUAGE, INDEX_SCRIPT, etc.
+    // Caution: the presence of "key|type" presents complications for refactoring.
+    // The row with "key|type" has four strings, while the others have three.
+    private static final String[][] NameTable = {
+        {"//ldml/localeDisplayNames/languages/language[@type=\"", "\"]", "language"},
+        {"//ldml/localeDisplayNames/scripts/script[@type=\"", "\"]", "script"},
+        {"//ldml/localeDisplayNames/territories/territory[@type=\"", "\"]", "territory"},
+        {"//ldml/localeDisplayNames/variants/variant[@type=\"", "\"]", "variant"},
+        {"//ldml/numbers/currencies/currency[@type=\"", "\"]/displayName", "currency"},
+        {"//ldml/numbers/currencies/currency[@type=\"", "\"]/symbol", "currency-symbol"},
+        {"//ldml/dates/timeZoneNames/zone[@type=\"", "\"]/exemplarCity", "exemplar-city"},
+        {"//ldml/dates/timeZoneNames/zone[@type=\"", "\"]/long/generic", "tz-generic-long"},
+        {"//ldml/dates/timeZoneNames/zone[@type=\"", "\"]/short/generic", "tz-generic-short"},
+        {"//ldml/dates/timeZoneNames/zone[@type=\"", "\"]/long/standard", "tz-standard-long"},
+        {"//ldml/dates/timeZoneNames/zone[@type=\"", "\"]/short/standard", "tz-standard-short"},
+        {"//ldml/dates/timeZoneNames/zone[@type=\"", "\"]/long/daylight", "tz-daylight-long"},
+        {"//ldml/dates/timeZoneNames/zone[@type=\"", "\"]/short/daylight", "tz-daylight-short"},
+        {"//ldml/localeDisplayNames/keys/key[@type=\"", "\"]", "key"},
+        {"//ldml/localeDisplayNames/types/type[@key=\"", "\"][@type=\"", "\"]", "key|type"},
+        {"//ldml/localeDisplayNames/subdivisions/subdivision[@type=\"", "\"]", "subdivision"},
+    };
```

Comment:
NameTable is moved from CLDRFile, otherwise unchanged. Further refactoring could be beneficial, to avoid the entanglement with integers INDEX_LANGUAGE, etc., but at least that entanglement is now encapsulated inside this class